carnival.cabal

```diff
@@ -146,6 +146,64 @@ executable         carnival
 
     ghc-options:       -threaded -O2 -rtsopts -with-rtsopts=-N
 
+executable carnival-integration
```

Comment:
Bummer that we need to repeat a lot of these options. Is there a way to specify `extension` or `ghc-options` for every target? Do we need all of these `build-depends` or are some of them only used from the library?